pandas/_libs/writers.pyx

```diff
@@ -125,19 +124,26 @@ def max_len_string_array(pandas_string[:] arr) -> Py_ssize_t:
 
     for i in range(length):
         val = arr[i]
-        if isinstance(val, str):
-            l = PyString_GET_SIZE(val)
-        elif isinstance(val, bytes):
-            l = PyBytes_GET_SIZE(val)
-        elif isinstance(val, unicode):
-            l = PyUnicode_GET_SIZE(val)
+        l = max_len_string(val)
 
         if l > m:
             m = l
 
     return m
 
 
+cpdef inline Py_ssize_t max_len_string(object val):
```

Comment:
Can you rename this function? It deals with more than just `strings` and returns the length, not the "maximum length". I think just `word_len` or something similar is fine

Reply:
Done, now `word_len`, also update the docstring text.